### PR TITLE
[RFC] Use JSONDecoder and Decodable instead of JSONSerialization

### PR DESCRIPTION
**Pull request by manicmaniac**
https://developer.apple.com/documentation/foundation/archives_and_serialization/encoding_and_decoding_custom_types

Swift 4 introduced `Decodable` protocol, which makes it easier for us to decode nested data structures like JSON.

I'm not sure whether the behavior of the entire program is correct or not, so please review it carefully :bow:

### CODE REVIEW

**Comment by manicmaniac** [comment]
```
$ swift run
Optional(app.AppQuery.Data(organizations: [app.Internal_1_Organizations(int: 1, float: 1.1, boolean: true, id: "a", name: "org")], nestedOrganizations: [Optional([Optional([app.Internal_2_Nestedorganizations(id: "a", name: "org", int: 1, float: 1.1, boolean: true)])])], outerAndInnerNullableOrganizations: nil, outerAndInnerNullableOrganizations2: Optional([nil, Optional(app.Internal_4_Outerandinnernullableorganizations2(id: "a", name: "org", int: 1, float: 1.1, boolean: true))]), outerNullableOrganizations: nil, outerNullableOrganizations2: Optional([app.Internal_6_Outernullableorganizations2(id: "a", name: "org", int: 1, float: 1.1, boolean: true)]), innerNullableOrganizations: [nil, Optional(app.Internal_7_Innernullableorganizations(id: "a", name: "org", int: 1, float: 1.1, boolean: true))], nullableOrganization: nil, organization: app.Internal_9_Organization(id: "a", name: "org", int: 1, float: 1.1, boolean: true)))
nil
Internal_1_Organizations(int: 1, float: 1.1, boolean: true, id: "a", name: "org")
Internal_1_Organizations(int: 1, float: 1.1, boolean: true, id: "a", name: "org")
Optional([Optional([app.Internal_2_Nestedorganizations(id: "a", name: "org", int: 1, float: 1.1, boolean: true)])])
```

File: src/index.ts [hunk 1]
```diff
@@ -292,30 +219,30 @@ export const plugin: PluginFunction = async (
...
-    init?(json: Any) {
-      guard let errors = json as? [String: Any] else {
-        return nil
-      }
+${indentMultiline(printStruct('Data', [], properties), 1)}
 
-      self.json = errors
+  struct Errors: Decodable {
+    init(from decoder: Decoder) throws {
+      // TODO
```
Comment (manicmaniac) [inline src/index.ts:245]: Need to define the structure of `errors` (basically `Decodable` doesn't support decoding an arbitrary dictionary).

**Comment by mtsmfm** [comment]
Sorry for the late reply, I'm reviewing

**Comment by mtsmfm** [comment]
Awesome!

**Pull request merged**